src/api/v1/utils.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * API specific utilities
+ */
+import * as Hapi from 'hapi';
+import * as Boom from 'boom';
+import { Response } from './schema/response';
+
+
+/*
+ * We only choose to re-format the boom response
+ * in order to get rid of the strange artefact of having
+ * an `error.error` attribute.
+ *
+ * This function can be done in one line with `renameKeys`
+ * but the compiler complains because it can't infer the
+ * correct keys
+ */
+export const formatBoom = ({ output: { payload } }: Boom<any>): Response => ({
+  error: {
+    statusCode: payload.statusCode,
+    type: payload.error,
+    message: payload.message,
+  },
+});
+
+/*
+ * If the response data is "raw", it is wrapped appropriately.
+ * Otherwise if the response has meta-data attached to it (and
+ * is therefore correctly formatted) it is simply passed through
+ */
+export const formatResponse = (res: Hapi.ResponseObject): Response => {
+  const r: any = res.source;
+
+  if (r.hasOwnProperty('data') && r.hasOwnProperty('meta')) {
```

Comment:
Do we have access to the request in this hook? If so I think it would make more sense to create metadata here (or in another hook) so that we don't have to do it individually in each handler.

Reply:
Agreed, but at the moment the metadata consists only of pagination information (see the type definition). Once there are other things of interest that aren't handler-specific, I'm definitely up for putting them here.

Reply:
One of the fields in `meta` is `total`, meaning the total number of records available to the client. How will the hook know about that?

Reply:
Good point, didn't think of that. I guess it will be easier to have a helper that builds this for us in queries.